src/settings.ts

```diff
@@ -0,0 +1,96 @@
+                if (debugSettingsRoot.settings.stepping.skipClasses) {
+                    stepFilters["skipClasses"] = await substituteFilterVariables(debugSettingsRoot.settings.stepping.skipClasses);
+                }
+                if (debugSettingsRoot.settings.stepping.skipSynthetics) {
+                    stepFilters["skipSynthetics"] = debugSettingsRoot.settings.stepping.skipSynthetics;
+                }
+                if (debugSettingsRoot.settings.stepping.skipStaticInitializers) {
+                    stepFilters["skipStaticInitializers"] = debugSettingsRoot.settings.stepping.skipStaticInitializers;
+                }
+                if (debugSettingsRoot.settings.stepping.skipConstructors) {
+                    stepFilters["skipConstructors"] = debugSettingsRoot.settings.stepping.skipConstructors;
+                }
+                extraSettings["stepFilters"] = stepFilters;
+            }
+            if (debugSettingsRoot.settings.exceptionBreakpoint && Object.keys(debugSettingsRoot.settings.exceptionBreakpoint).length) {
+                let exceptionFilters = {};
+                if (debugSettingsRoot.settings.exceptionBreakpoint.exceptionTypes) {
+                    exceptionFilters["exceptionTypes"] = debugSettingsRoot.settings.exceptionBreakpoint.exceptionTypes;
+                }
+                if (debugSettingsRoot.settings.exceptionBreakpoint.allowClasses) {
+                    exceptionFilters["allowClasses"] = debugSettingsRoot.settings.exceptionBreakpoint.allowClasses;
+                }
+                if (debugSettingsRoot.settings.exceptionBreakpoint.skipClasses) {
+                    exceptionFilters["skipClasses"] = await substituteFilterVariables(debugSettingsRoot.settings.exceptionBreakpoint.skipClasses);
+                }
+                extraSettings["exceptionFilters"] = exceptionFilters;
+                extraSettings["exceptionFiltersUpdated"] = true;
+            }
+
+            if (debugSettingsRoot.settings.jdwp) {
+                if (debugSettingsRoot.settings.jdwp.async) {
+                    extraSettings["asyncJDWP"] = debugSettingsRoot.settings.jdwp.async;
+                }
+                if (debugSettingsRoot.settings.jdwp.limitOfVariablesPerJdwpRequest) {
+                    extraSettings["limitOfVariablesPerJdwpRequest"] = Math.max(debugSettingsRoot.settings.jdwp.limitOfVariablesPerJdwpRequest, 1);
+                }
+                if (debugSettingsRoot.settings.jdwp.requestTimeout) {
+                    extraSettings["requestTimeout"] = Math.max(debugSettingsRoot.settings.jdwp.requestTimeout, 100);
+                }
+            }
+            const settings = await commands.executeCommand(Commands.JAVA_UPDATE_DEBUG_SETTINGS, JSON.stringify(
+                {
+                    ...debugSettingsRoot.settings,
+                    ...extraSettings,
+                    logLevel,
+                }));
+            if (logLevel === "FINE") {
```

Comment:
Since the `console.debug` goes to the coc logs, I'd prefer to rely on it's log level so let's remove this check against the `java.debug.logLevel`. See `:h :CocOpenLog`
```suggestion
```